### ISSUE TITLE
DeDRM failed on azw3 files downloaded from Amazon

### ISSUE THREAD

**Issue by reader45** [apprenticeharper/DeDRM_tools#1110]
Hello all

I downloaded two free Kindle books from Amazon.com the other day but Calibre could not remove the DRM. I haven't had problems before this and I have the latest version of the DRM tools installed. The files were downloaded for transfer to a physical Kindle via USB and they downloaded as azw3 files. Is this a known issue and if not, there any information I can supply to help out the people who maintain the tools?

Thanks all.

**Comment by ElleKayEm** [comment]
Do you have the E-ink Kindle's serial number in the DeDRM plugin customization?

**Comment by reader45** [comment]
Yes. I have been using Calibre successfully for a couple of years and have DeDRMed plenty of books before. These two books were the first that have come up with an error. Ironically, both were free to download from Amazon but they both came up with the error.

**Comment by ElleKayEm** [comment]
Post a log:

https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md#i-cannot-solve-my-problem-with-the-dedrm-plugin-and-now-i-need-to-post-a-log-how-do-i-do-that

**Comment by reader45** [comment]
H. P. Lovecraft: The Complete Collection https://www.amazon.com/gp/product/B078XHJL5Y/ref=kinw_myk_ro_title

calibre Debug log
calibre 3.44  embedded-python: True is64bit: True
Darwin-16.7.0-x86_64-i386-64bit Darwin ('64bit', '')
('Darwin', '16.7.0', 'Darwin Kernel Version 16.7.0: Wed Feb 27 00:29:57 PST 2019; root:xnu-3789.73.43~1/RELEASE_X86_64')
Python 2.7.15
OSX: ('10.12.6', ('', '', ''), 'x86_64')
Interface language: None
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 31, 0) && DeDRM (6, 7, 0) && Package KFX (from KFX Input) (1, 31, 0) && Kindle hi-res covers (0, 4, 0) && Set KFX metadata (from KFX Output) (1, 41, 1) && KFX Output (1, 41, 1) && Kindle covers (G) (0, 3, 0) && KFX metadata reader (from KFX Input) (1, 31, 0) && KFX Input (1, 31, 0) && Obok DeDRM (6, 7, 0) && ACE (1, 1, 0)
calibre 3.44  embedded-python: True is64bit: True
Darwin-16.7.0-x86_64-i386-64bit Darwin ('64bit', '')
('Darwin', '16.7.0', 'Darwin Kernel Version 16.7.0: Wed Feb 27 00:29:57 PST 2019; root:xnu-3789.73.43~1/RELEASE_X86_64')
Python 2.7.15
OSX: ('10.12.6', ('', '', ''), 'x86_64')
Interface language: None
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 31, 0) && DeDRM (6, 7, 0) && Package KFX (from KFX Input) (1, 31, 0) && Kindle hi-res covers (0, 4, 0) && Set KFX metadata (from KFX Output) (1, 41, 1) && KFX Output (1, 41, 1) && Kindle covers (G) (0, 3, 0) && KFX metadata reader (from KFX Input) (1, 31, 0) && KFX Input (1, 31, 0) && Obok DeDRM (6, 7, 0) && ACE (1, 1, 0)
devicePixelRatio: 1.0
logicalDpi: 72.0 x 72.0
physicalDpi: 129.000001937 x 129.000001937
Using calibre Qt style: True
[0.00] Starting up...
[0.12] Showing splash screen...
[0.40] splash screen shown
[0.40] Initializing db...
[1.37] db initialized
[1.38] Constructing main UI...
DEBUG:    0.0 obok::utilities.py - loading translations
DEBUG:    0.0 obok::dialogs.py - loading translations
DEBUG:    0.0 obok::config.py - loading translations
DEBUG:    0.2 obok::action_err.py - loading translations
Loaded cached store plugin for: Amazon Kindle at version: 17
Loaded cached store plugin for: Woblink at version: 15
Loaded cached store plugin for: Google Books at version: 6
[8.14] main UI initialized...
[8.14] Hiding splash screen
[8.15] splash screen hidden
[8.18] Started up in 8.18 seconds with 802 books
2020-06-02 00:06:04.080 calibre-debug[8009:1177141] Couldn't find or read strings file SlicesStrings
DeDRM v6.7.0: Trying to decrypt H. P. Lovecraft_ The Complete Collection.azw3
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v0.42.
Copyright © 2008-2017 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: H. P. Lovecraft: The Complete Collection
Found 2 keys to try after 1.1 seconds
Crypto Type is: 2
DeDRM v6.7.0: Failed to decrypt with error: No key found in 2 keys tried.
DeDRM v6.7.0: Looking for new default Kindle Key after 1.2 seconds
Found k4Mac kinf2011 file: /Users/dominic/Library/Application Support/Kindle/storage/.kinf2011
Decrypted key file using IDString '0d85c316aa42' and UserName 'dominic'
DeDRM v6.7.0: Found 1 new key
MobiDeDrm v0.42.
Copyright © 2008-2017 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: H. P. Lovecraft: The Complete Collection
Got DSN key from database default_key_1
Found 3 keys to try after 4.7 seconds
Crypto Type is: 2
DeDRM v6.7.0: Ultimately failed to decrypt after 4.7 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages/calibre/customize/ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 625, in run
  File "calibre_plugins.dedrm.__init__", line 575, in KindleMobiDecrypt
DeDRMError: DeDRM v6.7.0: Ultimately failed to decrypt after 4.7 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Added H. P. Lovecraft: The Complete Collection to db in: 1.1
Added 1 books in 22.3 seconds

**Comment by reader45** [comment]
Jane Austen: The Complete Novels
https://www.amazon.com/gp/product/B07Y5NN622/ref=kinw_myk_ro_title

calibre Debug log
calibre 3.44  embedded-python: True is64bit: True
Darwin-16.7.0-x86_64-i386-64bit Darwin ('64bit', '')
('Darwin', '16.7.0', 'Darwin Kernel Version 16.7.0: Wed Feb 27 00:29:57 PST 2019; root:xnu-3789.73.43~1/RELEASE_X86_64')
Python 2.7.15
OSX: ('10.12.6', ('', '', ''), 'x86_64')
Interface language: None
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 31, 0) && DeDRM (6, 7, 0) && Package KFX (from KFX Input) (1, 31, 0) && ACE (1, 1, 0) && KFX metadata reader (from KFX Input) (1, 31, 0) && KFX Input (1, 31, 0) && Kindle covers (G) (0, 3, 0) && Obok DeDRM (6, 7, 0) && Set KFX metadata (from KFX Output) (1, 41, 1) && KFX Output (1, 41, 1) && Kindle hi-res covers (0, 4, 0)
calibre 3.44  embedded-python: True is64bit: True
Darwin-16.7.0-x86_64-i386-64bit Darwin ('64bit', '')
('Darwin', '16.7.0', 'Darwin Kernel Version 16.7.0: Wed Feb 27 00:29:57 PST 2019; root:xnu-3789.73.43~1/RELEASE_X86_64')
Python 2.7.15
OSX: ('10.12.6', ('', '', ''), 'x86_64')
Interface language: None
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 31, 0) && DeDRM (6, 7, 0) && Package KFX (from KFX Input) (1, 31, 0) && ACE (1, 1, 0) && KFX metadata reader (from KFX Input) (1, 31, 0) && KFX Input (1, 31, 0) && Kindle covers (G) (0, 3, 0) && Obok DeDRM (6, 7, 0) && Set KFX metadata (from KFX Output) (1, 41, 1) && KFX Output (1, 41, 1) && Kindle hi-res covers (0, 4, 0)
devicePixelRatio: 1.0
logicalDpi: 72.0 x 72.0
physicalDpi: 129.000001937 x 129.000001937
Using calibre Qt style: True
[0.00] Starting up...
[0.02] Showing splash screen...
[0.19] splash screen shown
[0.19] Initializing db...
[0.23] db initialized
[0.23] Constructing main UI...
DEBUG:    0.0 obok::utilities.py - loading translations
DEBUG:    0.0 obok::dialogs.py - loading translations
DEBUG:    0.0 obok::config.py - loading translations
DEBUG:    0.0 obok::action_err.py - loading translations
Loaded cached store plugin for: Amazon Kindle at version: 17
Loaded cached store plugin for: Google Books at version: 6
Loaded cached store plugin for: Woblink at version: 15
[1.28] main UI initialized...
[1.28] Hiding splash screen
[1.29] splash screen hidden
[1.29] Started up in 1.29 seconds with 802 books
DeDRM v6.7.0: Trying to decrypt Jane Austen_ The Complete Novels.azw3
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v0.42.
Copyright © 2008-2017 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: Jane Austen: The Complete Novels
Found 2 keys to try after 0.1 seconds
Crypto Type is: 2
DeDRM v6.7.0: Failed to decrypt with error: No key found in 2 keys tried.
DeDRM v6.7.0: Looking for new default Kindle Key after 0.1 seconds
Found k4Mac kinf2011 file: /Users/dominic/Library/Application Support/Kindle/storage/.kinf2011
Decrypted key file using IDString '0d85c316aa42' and UserName 'dominic'
DeDRM v6.7.0: Found 1 new key
MobiDeDrm v0.42.
Copyright © 2008-2017 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: Jane Austen: The Complete Novels
Got DSN key from database default_key_1
Found 3 keys to try after 0.5 seconds
Crypto Type is: 2
DeDRM v6.7.0: Ultimately failed to decrypt after 0.5 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages/calibre/customize/ui.py", line 172, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 625, in run
  File "calibre_plugins.dedrm.__init__", line 575, in KindleMobiDecrypt
DeDRMError: DeDRM v6.7.0: Ultimately failed to decrypt after 0.5 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Added Jane Austen: The Complete Novels to db in: 0.2
Added 1 books in 1.4 seconds

**Comment by reader45** [comment]
If anybody can understand the logs, I'd love to know what they indicate.

And thanks ElleKayEm for your help.

**Comment by ElleKayEm** [comment]
Make sure you chose the correct E-ink Kindle (not a Fire) when downloading from the website.  Make sure you have entered the serial number correctly into the E-ink Kindle section of the DeDRM plugin customization.  Try copy and paste from the website.  You might want to update calibre too.

**Comment by reader45** [comment]
HI ElleKayEm

Thanks so much, that fixed it. I had recently deregistered one of my Kindles from Amazon and that was the Kindle that was still listed in Calibre. I was sure I had the right number but I misremembered. Entering the number for my other Kindle (the one still registered) fixed it. I haven't updated my Calibre because I was using an old Mac OS for other software, but I'm coming to the realisation that I'll probably have to update anyway and find another solution for the other software. How do I close this issue?
